File: restructure_projects.py
import json
import os

# Read the JSON file
file_path = os.path.join('Data', 'projects.json')
with open(file_path, 'r', encoding='utf-8') as f:
    data = json.load(f)

# Define the field order
field_order = [
    'project type',
    'title',
    'search',
    'year',
    'tags',
    'short description',
    'long description',
    'collaborators',
    'resources',
    'image_gallery',
    'image description',
    'image',
    'video description',
    'video'
]

# Process each project
processed_projects = []
for project in data:
    # Remove 'brief' field and other unwanted fields
    # Set default project type to "experiment" if missing or empty
    project_type = project.get('project type', '').strip() if project.get('project type') else ''
    if not project_type:
        project_type = 'experiment'
    
    # Create new project with ordered fields
    processed_project = {}
    for field in field_order:
        if field == 'project type':
            processed_project[field] = project_type
        elif field in ['tags', 'collaborators', 'resources', 'image', 'video']:
            # Arrays - use empty array if missing
            processed_project[field] = project.get(field, [])
        else:
            # Strings - use empty string if missing
            value = project.get(field)
            processed_project[field] = value if value is not None else ''
    
    processed_projects.append(processed_project)

# Write the restructured JSON
with open(file_path, 'w', encoding='utf-8') as f:
    json.dump(processed_projects, f, indent=2, ensure_ascii=False)

print(f'Processed {len(processed_projects)} projects')
print('Fields reordered and "brief" fields removed')

File: restructure_projects.js
const fs = require('fs');
const path = require('path');

// Read the JSON file
const filePath = path.join(__dirname, 'Data', 'projects.json');
const data = JSON.parse(fs.readFileSync(filePath, 'utf8'));

// Define the field order
const fieldOrder = [
  'project type',
  'title',
  'search',
  'year',
  'tags',
  'short description',
  'long description',
  'collaborators',
  'resources',
  'image_gallery',
  'image description',
  'image',
  'video description',
  'video'
];

// Process each project
const processedProjects = data.map(project => {
  // Remove 'brief' field
  delete project.brief;
  
  // Remove other fields not in the list (like 'role', 'links')
  const allowedFields = new Set(fieldOrder);
  Object.keys(project).forEach(key => {
    if (!allowedFields.has(key)) {
      delete project[key];
    }
  });
  
  // Set default project type to "experiment" if missing or empty
  if (!project['project type'] || project['project type'].trim() === '') {
    project['project type'] = 'experiment';
  }
  
  // Ensure all fields exist with default values
  const processedProject = {};
  fieldOrder.forEach(field => {
    if (field === 'tags' || field === 'collaborators' || field === 'resources' || field === 'image' || field === 'video') {
      // Arrays - use empty array if missing
      processedProject[field] = project[field] || [];
    } else {
      // Strings - use empty string if missing
      processedProject[field] = project[field] !== undefined && project[field] !== null ? project[field] : '';
    }
  });
  
  return processedProject;
});

// Write the restructured JSON
const output = JSON.stringify(processedProjects, null, 2);
fs.writeFileSync(filePath, output, 'utf8');

console.log(`Processed ${processedProjects.length} projects`);
console.log('Fields reordered and "brief" fields removed');
